pkg/controllers/provisioning/provisioner.go

```diff
@@ -158,7 +158,9 @@ func (p *Provisioner) launch(ctx context.Context, node *scheduling.Node) error {
 	// ourselves to enforce the binding decision and enable images to be pulled
 	// before the node is fully Ready.
 	if _, err := p.coreV1Client.Nodes().Create(ctx, k8sNode, metav1.CreateOptions{}); err != nil {
-		if !errors.IsAlreadyExists(err) {
+		if errors.IsAlreadyExists(err) {
+			logging.FromContext(ctx).Debugf("node %s already registered", k8sNode.Name)
```

Comment:
Added this log as it would have saved me a few hours this morning.  In a test if you somehow happen to get a duplicate node name, there are no errors reported since it could happen normally if kubelet comes up.  The end result is that your test doesn't fail with some "node already exists" error and if it does fail it will look like something is really wrong. In my case, a zonal topology spread was trying to schedule a node in "test-zone-2", the node already existed in "test-zone-1", no error occurred but the test for zone skew failed.